lua/cfc_time/server/storage_options/mysql.lua

```diff
@@ -207,9 +297,37 @@ function storage:GetTotalTime( steamID, callback )
     query:start()
 end
 
-function storage:CreateSession( callback, steamID, sessionStart, sessionEnd, duration )
-    local newSession = self:Prepare( "newSession", callback, steamID, sessionStart, sessionEnd, duration )
-    newSession:start()
+function storage:CreateSession( callback, steamID, sessionStart, sessionEnd, sessionDuration )
+    local maxDuration = self.MAX_SESSION_DURATION
+    local sessionsCount = math.ceil( maxDuration / sessionDuration )
+    if sessionsCount == math.huge then sessionsCount = 1 end
+
+    logger:debug( "[" .. tostring( steamID ) .. "] Creating " .. tostring( sessionsCount ) .. " sessions to accomodate duration of: " .. tostring( sessionDuration ) )
+
+    local transaction = self:InitTransaction()
+    transaction.onSuccess = function()
+        if callback then callback() end
+    end
```

Comment:
Could we not just do?
```suggestion
    transaction.onSuccess = callback
```